system_baseline/validators.py

```diff
@@ -46,6 +46,31 @@ def check_for_empty_name_values(facts):
             raise FactValidationError("value for %s cannot be empty" % fact["name"])
 
 
+def check_for_invalid_whitespace_name_values(facts):
+    """
+    check if any name or values have invalid whitespace at beginning and end; raises an exception.
+    """
+    for fact in facts:
+        if "values" in fact:
+            check_for_invalid_whitespace_name_values(fact["values"])
+        # if "name" in fact and re.search(r'^\s|\s$', fact["name"]):
+        if "name" in fact and (
+            fact["name"] != fact["name"].lstrip()
+            or fact["name"] != fact["name"].rstrip()
```

Comment:
you may want to break this out into its own method so you're not having to call it twice. like

```python
def _check_whitespace(input_string):
    if input_string == input_string.lstrip() == input_string.rstrip():
        return True
     return False
```

then you could rewrite your if statement to call that instead